### PR TITLE
Feature flag for containerd auditd rules

### LINKED ISSUE #556: [capi] Auditd produces too much logs
**What steps did you take and what happened:**

I built the latest CAPI Ubuntu image (qemu in this case, but this should affect all) and was using 
it for testing in ClusterAPI provider OpenStack. I saw that auditd produces a lot of logs. For example 
a simple cluster creation with a control plane node and 5 worker nodes produces about 200 MB of 
logs on the control plane node alone. I wasn't even running any tests on the cluster.

An example can be seen [here](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/782/pull-cluster-api-provider-openstack-make-conformance/1372667770679857152/artifacts/clusters/bootstrap/instances/conformance-kwla3r/cluster-cpnq16-control-plane-q28dh/systemd.log)
(I didn't catch it unfortunately before wasting a lot of space in the test buckets)

**What did you expect to happen:**

I was expecting to have some audit logs but maybe not that much :)

**Anything else you would like to add:**

I'm mostly opening this issue to ask if this is intended. I filter out the audit logs in the CAPO 
e2e tests now and in my production use cases I don't use the upstream auditd config. 

If there is a consensus we might configure auditd to produce a bit less logs.

**Environment:**

Project ([Image Builder for Cluster API](https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi):

Additional info for Image Builder for Cluster API related issues:

- OS (e.g. from `/etc/os-release`, or `cmd /c ver`):
```bash
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Cluster-api version (if using): latest ClusterAPI Provider OpenStack but this shouldn't matter
- Kubernetes version: (use `kubectl version`): Kubernetes v1.20.4

/kind bug

### PR DESCRIPTION
**Pull request by knabben**
What this PR does / why we need it:
Containerd auditd rules are verbose and can cause high disk latency when the node is on high usage. Adding a feature flag to toggle this option `enable_containerd_audit`, not enabled by default. 

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): 
Fixes #556

**Additional context**
Tests were made with node under pressure and enabling containerd trace for 100s can cause delay on disk IO.

### CODE REVIEW

**Comment by sbueringer** [comment]
@knabben fixes https://github.com/kubernetes-sigs/image-builder/issues/556?

File: images/capi/ansible/roles/node/defaults/main.yml [hunk 1]
```diff
@@ -116,3 +116,6 @@ leak_local_mdns_to_dns: false
 build_target: "virt"
 cloud_cfg_file: "/etc/cloud/cloud.cfg"
 external_binary_path: "{{ '/opt/bin' if ansible_os_family == 'Flatcar' else '/usr/local/bin' }}"
+
+# Enable containerd trace audit in auditd, default: false.
+enable_containerd_audit: false
```
Comment (kkeshavamurthy) [inline images/capi/ansible/roles/node/defaults/main.yml:121]: Can this be added to the [ansible-args.json](https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/packer/config/ansible-args.json#L3) file as well pls.

Comment (knabben) [reply images/capi/ansible/roles/node/defaults/main.yml:121]: @kkeshavamurthy does it need to go into `packer/ova/packer-node.json` or this is already setting the default?

**Comment by kkeshavamurthy** [comment]
@knabben Can you pls rebase this. We'll get this merged.

**Comment by mboersma** [comment]
@knabben sorry this has dangled so long. If you can rebase it, I think it's ready to merge.

**Pull request merged**